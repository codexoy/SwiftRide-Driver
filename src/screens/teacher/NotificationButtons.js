import React, { Fragment } from "react"
import { StyleSheet, View, Text, TouchableHighlight } from "react-native"
import { strings } from "../../i18n"

export default class NotificationButtons extends React.Component {
	render() {
		return (
			<Fragment>
				<TouchableHighlight testID="approve" style={styles.touchable}>
					<View
						style={{
							...styles.button,
							...{ backgroundColor: "rgb(12,116,244)" }
						}}
					>
						<Text style={styles.buttonText}>
							{strings("approve")}
						</Text>
					</View>
				</TouchableHighlight>
				<TouchableHighlight style={styles.touchable}>
					<View
						style={{
							...styles.button,
							...{ backgroundColor: "rgb(197,197,197)" }
						}}
					>
						<Text style={styles.buttonText}>{strings("edit")}</Text>
					</View>
				</TouchableHighlight>
				<TouchableHighlight style={styles.touchable}>
					<View
						style={{
							...styles.button,
							...{ backgroundColor: "rgb(240,8,48)" }
						}}
					>
						<Text style={styles.buttonText}>
							{strings("postpone")}
						</Text>
					</View>
				</TouchableHighlight>
			</Fragment>
		)
	}
}

const styles = StyleSheet.create({
	touchable: {
		flex: 0.33
	},
	button: {
		alignItems: "center",
		padding: 8,
		borderRadius: 3,
		backgroundColor: "#000"
	},
	buttonText: {
		fontSize: 14,
		color: "#fff",
		fontWeight: "bold"
	}
})
