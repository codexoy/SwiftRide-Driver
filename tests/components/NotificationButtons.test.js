import React from "react"

import renderer from "react-test-renderer"

import NotificationButtons from "../../src/screens/teacher/NotificationButtons"

describe("NotificationButtons", () => {
	test("view renders correctly", () => {
		const tree = renderer
			.create(
				<NotificationButtons
					style={{ height: 30 }}
					name="test"
					type="new_lesson"
					date="test"
					hours="test-test"
				/>
			)
			.toJSON()
		expect(tree).toMatchSnapshot()
	})
})
