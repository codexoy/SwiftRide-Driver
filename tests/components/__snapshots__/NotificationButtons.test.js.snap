// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationButtons view renders correctly 1`] = `
Array [
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flex": 0.49,
        "opacity": 1,
      }
    }
    testID="approve"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgb(12,116,244)",
          "borderRadius": 3,
          "padding": 8,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#fff",
            "fontSize": 14,
            "fontWeight": "bold",
          }
        }
      >
        אישור
      </Text>
    </View>
  </View>,
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flex": 0.49,
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgb(240,8,48)",
          "borderRadius": 3,
          "padding": 8,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#fff",
            "fontSize": 14,
            "fontWeight": "bold",
          }
        }
      >
        ביטול
      </Text>
    </View>
  </View>,
]
`;
